### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/DnaChisel/commit/df5c1ec886b3b8b02f93bb63e8c14a1ad01f0350#commitcomment-45566687

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/DnaChisel/commit/a0e31fdbf1937feb1e1fc58d3d1f7a0ee186dcfa

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: dnachisel/SequencePattern/DnaNotationPattern.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class DnaNotationPattern(SequencePattern):
     """Class for patterns in plain DNA notation: ATTGCCA, GCNNKTA, etc.
 
-    If the sequence is not palyndromic, the pattern will be looked for in
+    If the sequence is not palindromic, the pattern will be looked for in
```
Comment (jsoref) [inline dnachisel/SequencePattern/DnaNotationPattern.py:13]: [Google Ngram Viewer](https://books.google.com/ngrams/graph?content=palyndromic%2Cpalindromic&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cpalyndromic%3B%2Cc0%3B.t1%3B%2Cpalindromic%3B%2Cc0)

File: docs/ref/builtin_specifications.rst [hunk 2]
```diff
@@ -24,7 +24,7 @@ Summary
     HarmonizeRCA
     EnforceGCContent
     EnforceMeltingTemperature
-    EnforcePatternOccurence
+    EnforcePatternOccurrence
```
Comment (jsoref) [inline docs/ref/builtin_specifications.rst:27]: It's possible this is an API change

File: dnachisel/reports/optimization_reports.py [hunk 3]
```diff
@@ -290,7 +290,7 @@ def plot_optimization_changes(problem):
 def write_optimization_report(
     target,
     problem,
-    project_name="unnammed",
+    project_name="unnamed",
```
Comment (jsoref) [inline dnachisel/reports/optimization_reports.py:293]: This is probably a user visible change

**Review by jsoref — COMMENTED** [review]
I can split/drop things if requested

**Comment by Zulko** [comment]
Oh wow that's a lot of typos, thanks @jsoref for this spell-checking tool! @veghp sorry for the typos. Most corrections in this PR look good, but it looks like I have consistently written `occurence` with a single `r`, and unfortunately fixing that will require an API change + version bump (to correct class `EnforcePatternOccurence` to `EnforcePatternOccurrence`).

**Comment by jsoref** [comment]
This should fix the build -- I have tooling for renames, but apparently not for git, and I haven't integrated renaming into my tooling...

As API bumps are major things, I'm going to drop it from this PR. I could try to make the actual PR for it, but as I'm not precisely sure what it'll entail, it might make more sense for me to give you some commits and let you write the API compat bits you require.

File: docs/ref/builtin_specifications.rst (hunk 2 above)
Comment (jsoref) [reply docs/ref/builtin_specifications.rst]: I've moved this to https://github.com/jsoref/DnaChisel/tree/spelling-occurrence

**Comment by veghp** [comment]
Thank you for these corrections. I reviewed most of the other packages for spelling, but not this one. I merged your commits into the [`dev` branch](https://github.com/Edinburgh-Genome-Foundry/DnaChisel/network) and they will be included in the next release, except for the change in the spelling of 'palindromic'. I stored that in branch `palindromic`, and will be applied together with other API changes in the [next major release](https://github.com/Edinburgh-Genome-Foundry/DnaChisel/projects/1).
As the change to default value "unnamed" affects the name of the report only, I left that in.

Please note that the tool introduced one spelling error ('respectivelyy', which I corrected):
https://github.com/Edinburgh-Genome-Foundry/DnaChisel/pull/48/files#diff-2d7660d970eabc63831ad1b44104bc87976f5f6d15acc187354d3a1aef578d35R40

**Comment by jsoref** [comment]
oops. I'm still working on improving those edges. I actually added a thing to my tool between this PR and today to try to avoid eating my own tail.

**Comment by veghp** [comment]
Sounds good. I'll close this PR as the commits are already in the dev and palindromic branches.

**Comment by jsoref** [comment]
@veghp: note that https://github.com/jsoref/DnaChisel/tree/spelling-occurrence is currently floating outside as @Zulko identified it as problematic and I excluded it from this PR. (Maybe I shouldn't have.)

**Comment by veghp** [comment]
Yes, thanks, it has been noted in the [todo list](https://github.com/Edinburgh-Genome-Foundry/DnaChisel/projects/1). We will do this change "manually" with an IDE, as DNA Chisel has many dependents.

**Pull request closed without merging**